pallets/collator-assignment/src/lib.rs

```diff
@@ -121,26 +125,66 @@ pub mod pallet {
         /// collators should be queued collators
         pub fn assign_collators(
             current_session_index: &T::SessionIndex,
-            collators: Vec<T::AccountId>,
+            random_seed: [u8; 32],
+            mut collators: Vec<T::AccountId>,
         ) -> SessionChangeOutcome<T> {
             // We work with one session delay to calculate assignments
             let session_delay = T::SessionIndex::one();
             let target_session_index = current_session_index.saturating_add(session_delay);
             // We get the containerChains that we will have at the target session
-            let container_chain_ids =
+            let mut container_chain_ids =
                 T::ContainerChains::session_container_chains(target_session_index);
+
+            // If the random_seed is all zeros, we don't shuffle the list of collators nor the list
+            // of container chains.
+            // This should only happen in tests, and in the genesis block.
+            if random_seed != [0; 32] {
+                let mut rng: ChaCha20Rng = SeedableRng::from_seed(random_seed);
+                collators.shuffle(&mut rng);
+                container_chain_ids.shuffle(&mut rng);
+            }
+
             // We read current assigned collators
             let old_assigned = Self::read_assigned_collators();
             // We assign new collators
             // we use the config scheduled at the target_session_index
-            let new_assigned = Self::assign_collators_always_keep_old(
-                collators,
-                &container_chain_ids,
-                T::HostConfiguration::min_collators_for_orchestrator(target_session_index) as usize,
-                T::HostConfiguration::max_collators_for_orchestrator(target_session_index) as usize,
-                T::HostConfiguration::collators_per_container(target_session_index) as usize,
-                old_assigned.clone(),
-            );
+            let new_assigned =
+                if T::ShouldRotateAllCollators::should_rotate_all_collators(target_session_index) {
```

Comment:
Questioning myself whether it makes sense that we additionally have a storage item as a switch-kill method for the rotation in this pallet. What do you think? does it break many things?

Reply:
A kill-switch for the method should_rotate_all_collators can be implemented directly in the runtime, for example the storage item could live in pallet_configuration.